ga4gh/datamodel/variants.py

```diff
@@ -680,20 +641,14 @@ def convertVariant(self, record, callSetIds):
                     value = value.split(',')
                 variant.info[key] = _encodeValue(value)
 
-        # NOTE: THE LABELED LINES SHOULD BE REMOVED ONCE PYSAM SUPPORTS
-        # phaseset
-
-        sampleData = record.__str__().split()[9:]  # REMOVAL
         variant.calls = []
-        sampleIterator = 0  # REMOVAL
-        if callSetIds is not None:
-            for name, call in record.samples.iteritems():
-                if self.getCallSetId(name) in callSetIds:
-                    genotypeData = sampleData[sampleIterator].split(
-                        ":")[0]  # REMOVAL
-                    variant.calls.append(self._convertGaCall(
-                        record.id, name, call, genotypeData))  # REPLACE
-                sampleIterator += 1  # REMOVAL
+        if callSetIds is not None and len(callSetIds) > 0:
+            for callSetId in callSetIds:
+                callSet = self.getCallSet(callSetId)
```

Comment:
Why would callSet be None here? This seems like an error condition to me. We shouldn't silently hide error conditions, this will cause very hard to find bugs. Much better to fail loudly with an exception.